Regional packages

kbevers opened the pull request:
In response to #7.

Still need to create a `-europe` package. Sharing now to get feedback on the general idea before adding the European package.

kbevers committed:
Move Australian grid descriptions to README.OCEANIA

kbevers committed:
Created regional package for North America including gvr2016.gtx, new init file GL and new grid gvr2000.gtx.

sebastic commented on north-america/README.NORTHAMERICA:
```diff
@@ -0,0 +1,51 @@
+For an installed PROJ this may be /usr/local/share/proj or /usr/share/proj
+on unix style operating systems.
+
+## About the Oceania PROJ ressource package
```
s/Oceania/North America/

sebastic commented on north-america/README.NORTHAMERICA:
```diff
@@ -0,0 +1,51 @@
+
+## About the Oceania PROJ ressource package
+
+The Oceania PROJ ressource package is a collections of grids
```
s/Oceania/North America/

sebastic commented on oceania/README.OCEANIA:
```diff
@@ -4,8 +4,49 @@ The files in this package can be unpacked in the PROJ data directory.
 
-## About the Oceania Datum Grid package
+## About the Oceania PROJ ressource package
 
```
s/ressource/resource/

sebastic commented on oceania/README.OCEANIA:
```diff
@@ -4,8 +4,49 @@ The files in this package can be unpacked in the PROJ data directory.
 
-Due to the large size of the Australian grid files, these files are
-distributed separately.
+The Oceania PROJ ressource package is a collections of grids
```
s/ressource/resource/

sebastic commented on oceania/README.OCEANIA:
```diff
@@ -4,8 +4,49 @@ The files in this package can be unpacked in the PROJ data directory.
+*Source*: [ICSM](https://github.com/icsm-au/transformation_grids)  
+*Format*: NTv2  
+*License*: [Creative Commons Attribution 4.0]((https://creativecommons.org/licenses/by/4.0/))
+
```
Double parentheses should be replaced with single.

sebastic commented on oceania/README.OCEANIA:
```diff
@@ -4,8 +4,49 @@ The files in this package can be unpacked in the PROJ data directory.
+
+### Australia: AGD66 -> GDA94
+
+*Source*: [ICSM](https://github.com/icsm-au/transformation_grids)  
```
Since markup is now used, we should apply the `.github` symlink trick to have it rendered on GitHub.

sebastic commented on north-america/CMakeLists.txt:
```diff
@@ -0,0 +1,16 @@
+set(CPACK_SOURCE_GENERATOR "TGZ;ZIP")
+set(CPACK_INCLUDE_TOPLEVEL_DIRECTORY 0)
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "proj-datumgrid-north-america-${PROJ_DATUMGRID_NORTHAMERICA_VERSION_MAJOR}.${PROJ_DATUMGRID_NORTHAMERICA_VERSION_MINOR}")
+set(CPACK_SOURCE_IGNORE_FILES "/build/;/cmake/;CMakeLists.txt;.swp$;.*~;${CPACK_SOURCE_IGNORE_FILES}")
```
When the `.github` symlink trick is used for the README rendering, the `.github` directory needs to be added the the `CPACK_SOURCE_IGNORE_FILES`.

sebastic commented on north-america/README.NORTHAMERICA:
```diff
@@ -0,0 +1,51 @@
+* gvr2016.gtx
+
+### Greenland: GRS80 ellipsoidal heights -> GVR2000
+#
```
s/#//

sebastic commented:
The `north-america` directory still needs to be ignored in the main `CMakeLists.txt`:
```
Built target dist
-rw-r--r-- bas/bas      318464 2005-05-19 13:10 nzgd2kgrid0005.gsb
-rw-r--r-- bas/bas        5190 2018-02-26 19:51 README.DATUMGRID
-rw-r--r-- bas/bas      277424 2009-09-23 09:56 ntf_r93.gsb
-rw-r--r-- bas/bas       26728 2018-02-26 19:54 prvi
-rw-r--r-- bas/bas      452008 2018-02-26 19:54 hawaii
-rw-r--r-- bas/bas        4424 2018-02-26 19:54 TN
-rw-r--r-- bas/bas       26728 2018-02-26 19:54 stlrnc
-rw-r--r-- bas/bas         232 2018-02-26 19:54 null
-rw-r--r-- bas/bas       11112 2018-02-26 19:54 WO
-rw-r--r-- bas/bas       59208 2018-02-26 19:54 stgeorge
drwxr-xr-x bas/bas           0 2018-02-26 19:54 north-america/
-rw-r--r-- bas/bas        1895 2018-02-26 19:51 north-america/README.NORTHAMERICA
-rw-r--r-- bas/bas    18930252 2018-02-26 17:50 north-america/gvr2016.gtx
-rw-r--r-- bas/bas        3720 2018-02-26 17:50 north-america/GL
-rw-r--r-- bas/bas     1304644 2018-02-26 17:50 north-america/gvr2000.gtx
-rw-r--r-- bas/bas        6760 2018-02-26 19:54 WI
-rw-r--r-- bas/bas     4153000 2018-02-26 15:21 egm96_15.gtx
-rw-r--r-- bas/bas       83696 2018-02-14 11:49 BETA2007.gsb
-rw-r--r-- bas/bas        7048 2018-02-26 19:54 stpaul
-rw-r--r-- bas/bas     1113184 2001-04-05 12:45 ntv1_can.dat
-rw-r--r-- bas/bas        3560 2018-02-26 19:54 MD
-rw-r--r-- bas/bas      964044 2018-02-20 11:30 dvr90.gtx
-rw-r--r-- bas/bas      264424 2018-02-26 19:54 conus
-rw-r--r-- bas/bas     1053928 2018-02-26 19:54 alaska
-rw-r--r-- bas/bas        8872 2018-02-26 19:54 FL
-rw-r--r-- bas/bas      643244 2018-02-20 11:30 fvr09.gtx
```

kbevers commented:
I'm on it! Thanks.

sebastic commented on README.DATUMGRID:
```diff
@@ -197,18 +139,33 @@ grid coordinates are referenced to ETRS89.
 * fvr09.gtx
 
-*Package*: proj-datumgrid
+## Regional PROJ ressource packages
```
s/ressource/resource/

sebastic commented on README.DATUMGRID:
```diff
@@ -197,18 +139,33 @@ grid coordinates are referenced to ETRS89.
+
+### North America
+
+The ressource package for Oceania at the moment includes grids from the
```
s/ressource/resource/

sebastic replied:
s/Oceania/North America/

sebastic commented on README.DATUMGRID:
```diff
@@ -197,18 +139,33 @@ grid coordinates are referenced to ETRS89.
+### Oceania
 
-* gvr2016.gtx
+The ressource package for Oceania at the moment includes grids from the
```
s/ressource/resource/

sebastic commented on README.DATUMGRID:
```diff
@@ -197,18 +139,33 @@ grid coordinates are referenced to ETRS89.
+
+### North America
+
+The ressource package for Oceania at the moment includes grids from the
```
Wouldn't "includes grids *for* the following countries" be better?

kbevers replied:
Yes. I am changing it to "The regional package for <REGION> include resources for the following countries:"

kbevers committed:
Added regional package for Europe.

Includes grids and init files covering Denmark, The Faroe Island
and the member nations of The Nordic Geodetic Commision.

kbevers commented:
European package added as well now.

sebastic commented on README.DATUMGRID:
```diff
@@ -171,44 +117,45 @@ NAD83 to NAD83.
+### Europe
 
-*Package*: proj-datumgrid
+The regional package for Europe include resources for the following
```
s/include/&s/

sebastic commented on README.DATUMGRID:
```diff
@@ -171,44 +117,45 @@ NAD83 to NAD83.
+### North America
 
-*Package*: proj-datumgrid
+The regional package for North America include resources for the following
```
s/include/&s/

sebastic commented on README.DATUMGRID:
```diff
@@ -171,44 +117,45 @@ NAD83 to NAD83.
+### Oceania
 
-* gvr2016.gtx
+The regional package for Oceania include resources for the following
```
s/include/&s/

sebastic commented on europe/README.EUROPE:
```diff
@@ -0,0 +1,103 @@
+
+### Nordic+Baltic countries: ITRFxx -> Local ETRS89 realizations
+
+*Source*: [The Nordic Geodetic Commision](https://github.com/NordicGeodesy/NordicTransformations)  
```
s/Commision/Commission/, unless the typo is part of the offical name.

kbevers replied:
Nah, luckily it's just my typo :)

kbevers committed:
Fix typos in READMEs

sebastic commented on README.DATUMGRID:
```diff
@@ -171,44 +117,45 @@ NAD83 to NAD83.
-*Package*: proj-datumgrid
-
-### Denmark: GRS80 ellipsoidal heights -> DVR90 heights
+# Regional PROJ ressource packages
```
s/ressource/resource/

sebastic commented:
The "Some large grid files are distributed in separate packages." line in README.DATUMGRID can probably be removed since we have a whole section about that now.

sebastic commented on README.DATUMGRID:
```diff
@@ -171,44 +117,45 @@ NAD83 to NAD83.
 
-Transform ellipsoidal heights to the local height system FVR09. The horizontal
-grid coordinates are referenced to ETRS89.
 
```
The two empty lines can be collapsed into a single empty line.

sebastic commented on README.DATUMGRID:
```diff
@@ -33,48 +33,6 @@ Some large grid files are distributed in separate packages.
-* GDA94_GDA2020_conformal_and_distortion.gsb
-
-*Package*: proj-datumgrid-oceania
 
```
The two empty lines can be collapsed into a single empty line.

sebastic commented on europe/README.EUROPE:
```diff
@@ -0,0 +1,103 @@
+ PROJ-DATUMGRID-EUROPE
```
The `#` at the start of the line got lost

kbevers commented:
I hope that I have fixed all the typos now :-)

sebastic commented:
Looks like it, both my eyeballs and `codespell` no longer find any.

The content of the tarballs looks good too.

Should we explicitly document the release procedure for the other regions in `HOWTORELEASE`? Or make that process less explicit to apply to all regions?

kbevers commented:
> Should we explicitly document the release procedure for the other regions in HOWTORELEASE? Or make that process less explicit to apply to all regions?

I think it is fine to just have an example of how it is done. The Oceania section is good enough for me at least.

sebastic commented:
> I think it is fine to just have an example of how it is done. The Oceania section is good enough for me at least.

You just can't copy/paste the `git tag` oneliner :-)

kbevers commented:
> You just can't copy/paste the git tag oneliner :-)

That is true. To be honest I would probably just write it manually anyway. But the oneliner is very impressive and well above my command line abilities!
